### PR TITLE
Workflow Error Page

### PR DESCRIPTION
**Pull request by Alex-Tideman**
## Description & motivation 💭 
Previously all we did was show a spinner and throw a toast notification for 404/other errors. This was very difficult to read and didn't last long. It also led to confusion on if the workflow was still loading.

This is also better than our default Error component because it allows the breadcrumb Back to Workflows and gives more humanized errors.

I added custom onError handling for fetchWorkflow and a new WorkflowError component to handle the error.

### Screenshots (if applicable) 📸 
Screen Shot 2024-01-24 at 1 07 17 PM

### How was this tested 👻 

- [x] Manual testing
- [ ] E2E tests added
- [ ] Unit tests added

### CODE REVIEW

**Review by rossedfort — APPROVED** [review]
Nice improvement, I'll let design weigh in but my only suggestion would maybe be make the error text smaller and omit the logo, or move the logo above the text

File: src/lib/components/workflow/workflow-error.svelte [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      icon="chevron-left"
+    >
+      {translate('workflows.back-to-workflows')}
+    </Link>
+  </div>
+  <h1
+    data-testid="workflow-error-heading"
+    class="overflow-hidden text-3xl font-medium"
+  >
+    {error.statusText}
```
Comment (laurakwhit) [inline src/lib/components/workflow/workflow-error.svelte:37]: What do you think about making this a little more similar to the existing `Error` page and have the status text centered with the other info. ⬇️ 

Screenshot 2024-01-24 at 11 32 11 AM

File: src/lib/services/workflow-service.ts [hunk 2]
```diff
@@ -206,17 +207,28 @@ export const fetchAllArchivedWorkflows = async (
 export async function fetchWorkflow(
   parameters: GetWorkflowExecutionRequest,
   request = fetch,
-): Promise<WorkflowExecution> {
+): Promise<{
+  workflow: WorkflowExecution | undefined;
```
Comment (laurakwhit) [inline src/lib/services/workflow-service.ts:211]: Noticed that workflow or error doesn't actually show up as possibly `undefined` ⬇️  Wondering if there's a better way to type this so it does...

Screenshot 2024-01-24 at 11 34 35 AM

Also wondering if we shouldn't just return the object without the `undefined` property (e.g. `{ error: e }`.

**Review by laurakwhit — APPROVED** [review]
Some small comments r.e. types and design, but otherwise lgtm

File: src/lib/components/workflow/workflow-error.svelte (hunk 1 above)
Comment (Alex-Tideman) [reply src/lib/components/workflow/workflow-error.svelte]: 👍 updated

File: src/lib/services/workflow-service.ts (hunk 2 above)
Comment (Alex-Tideman) [reply src/lib/services/workflow-service.ts]: Good call

**Pull request merged**